Tasks/Npm/Tests/NpmMockHelper.ts

```diff
@@ -0,0 +1,67 @@
+import ma = require('vsts-task-lib/mock-answer');
+import tmrm = require('vsts-task-lib/mock-run');
+
+export class NpmMockHelper {
+    static NpmCmdPath = "C:\\Program Files (x86)\\nodejs\\npm";
+    static FakeWorkingDirectory = "fake/wd";
+
+    public answers: ma.TaskLibAnswers = {};
+    
+    constructor(
+        private tmr: tmrm.TaskMockRunner,
+        public command: string,
+        public args: string) { 
+        process.env['AGENT_HOMEDIRECTORY'] = "c:\\agent\\home\\directory";
+        process.env['BUILD_SOURCESDIRECTORY'] = "c:\\agent\\home\\directory\\sources";
+        process.env['ENDPOINT_AUTH_SYSTEMVSSCONNECTION'] = "{\"parameters\":{\"AccessToken\":\"token\"},\"scheme\":\"OAuth\"}";
+        process.env['ENDPOINT_URL_SYSTEMVSSCONNECTION'] = "https://example.visualstudio.com/defaultcollection";
+        process.env['SYSTEM_DEFAULTWORKINGDIRECTORY'] = "c:\\agent\\home\\directory";
+        process.env['SYSTEM_TEAMFOUNDATIONCOLLECTIONURI'] = "https://example.visualstudio.com/defaultcollection";
+
+        tmr.setInput('cwd', NpmMockHelper.FakeWorkingDirectory);
+        tmr.setInput('command', command);
+        tmr.setInput('arguments', args);
+
+        this.setDefaultAnswers();
+    }
+
+    public run(result: ma.TaskLibAnswerExecResult = null) {
+        if (result) {
+            this.setExecResponse(result);
+        }
+        this.tmr.setAnswers(this.answers);
+        this.tmr.run();
+    }
+
+    public useDeprecatedTask() {
+        process.env['USE_DEPRECATED_TASK_VERSION'] = 'true';
```

Comment:
Nit: Should we use a constant for this. Just thinking out loud, since any change to this constant would need change in multiple places.